File: aggregator/src/main/java/com/web3analytics/models/LiquidityAnalytics.java
package com.web3analytics.models;

/**
 * Liquidity behavior analytics computed from Mint/Burn/Transfer events.
 * Transfer events are correlated by transactionHash to compute LP token amounts.
 */
public record LiquidityAnalytics(
        String windowId,
        long windowStart,
        long windowEnd,
        String pairAddress,
        String token0Symbol,
        String token1Symbol,
        int mintCount,
        int burnCount,
        String totalMintAmount0,
        String totalMintAmount1,
        String totalBurnAmount0,
        String totalBurnAmount1,
        String netLiquidityChange0,
        String netLiquidityChange1,
        String totalLpTokensMinted,
        String totalLpTokensBurned,
        String netLpTokenChange,
        int transferCount,
        int uniqueMintProviders,
        int uniqueBurnProviders,
        int uniqueProviders,
        int churnedProviders,
        long processedAt
) {
}

File: analytics-service/src/main/kotlin/com/web3analytics/analytics/model/LiquidityAnalytics.kt
package com.web3analytics.analytics.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@JsonIgnoreProperties(ignoreUnknown = true)
data class LiquidityAnalytics(
    val windowId: String,
    val windowStart: Long,
    val windowEnd: Long,
    val pairAddress: PairAddress,
    val token0Symbol: String? = null,
    val token1Symbol: String? = null,
    val mintCount: Int = 0,
    val burnCount: Int = 0,
    val totalMintAmount0: String? = null,
    val totalMintAmount1: String? = null,
    val totalBurnAmount0: String? = null,
    val totalBurnAmount1: String? = null,
    val netLiquidityChange0: String? = null,
    val netLiquidityChange1: String? = null,
    val totalLpTokensMinted: String? = null,
    val totalLpTokensBurned: String? = null,
    val netLpTokenChange: String? = null,
    val transferCount: Int = 0,
    val uniqueMintProviders: Int = 0,
    val uniqueBurnProviders: Int = 0,
    val uniqueProviders: Int = 0,
    val churnedProviders: Int = 0,
    val processedAt: Long = 0
)
